### PR TITLE
Bonding Contracts UI enhancement

### LINKED ISSUE #9: Clarify Withdraw Shares Explaination
image
> why doesn't my staked LP tokens show up?
> it says i have 0 bonding shares
> 0 that are withdrawable
> but perhaps we should adjust this to be more clear

### PR DESCRIPTION
**Pull request by Zequez**
Captura de Tela 2021-09-30 às 19 18 10
Captura de Tela 2021-09-30 às 19 17 57
`
Captura de Tela 2021-09-30 às 19 17 52
Captura de Tela 2021-09-30 às 19 17 38
Captura de Tela 2021-09-30 às 19 17 08
Captura de Tela 2021-09-30 às 19 16 09

The following components are deprecated by this new UI:

- `chefugov.tsx` (deleted): the "claim Ubiquity rewards" button, it only worked with the first user share
- `deposit.share.redeem.tsx` (deleted): It was just broken, didn't even attempted to call the removeLiquidity function
- `deposit.share.balance.tsx` (deleted): The information is loaded on the BondingSharesExplorer component
- `deposit.share.tsx -> DepositShare.tsx`: Integrated into BondingSharesExplorer

All the features are quite self-explanatory.

Resolves ubiquity/uad-contracts#9  ubiquity/uad-contracts#6 and all the issues community members were having with seeing their bonding shares. Also solves this issue from the uad-contracts repo ubiquity/ubiquity-dollar-development#122

Calculates the APY too, so resolves this one https://github.com/pavlovcik/uad-ui/issues/4

### CODE REVIEW

**Comment by 0x4007** [comment]
#### 📝 Changed routes:


| [**/monitor**](https://uad-ui-do2ipzdd1-pavlovcik.vercel.app/monitor) | [**/styles/tailwind**](https://uad-ui-do2ipzdd1-pavlovcik.vercel.app/styles/tailwind) |
|:-:|:-:|
|<a href="https://uad-ui-do2ipzdd1-pavlovcik.vercel.app/monitor">Screenshot of /monitor</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/wUPTj13CZm7mAA7HmT3dXOiLhu8RdE8VZ8oRZP7Ff-eQG-6v-yN7bTdak7E0MAxWZmtyWFdDwVsM1A4pIGTAbA.png">(view full size)</a> |<a href="https://uad-ui-do2ipzdd1-pavlovcik.vercel.app/styles/tailwind">Screenshot of /styles/tailwind</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/WdLYOczlshdbhUHL5FvqGsFZe7cpG8qhnY5nVt2Aa-6iG_Ax-imGpxmyhVwXKYPDZF4Jw_-nB70d1tkIJa5mFA.png">(view full size)</a> |







Commit 4e6f568e8d7d0e159f02bf48549ae5e15770b956 (https://uad-ui-do2ipzdd1-pavlovcik.vercel.app).

**Comment by Zequez** [comment]
Looks like it's working on the current deployment; merge?

File: pages/styles/index.css [hunk 1]
```diff
@@ -429,7 +437,7 @@ a:-webkit-any-link {
 }
 
 .bg-blur {
-  backdrop-filter: blur(24px);
+  /* backdrop-filter: blur(24px); */
```
Comment (0x4007) [inline pages/styles/index.css]: @Zequez you can have this disabled on local if you want but on prod I'd like it on

File: pages/styles/index.css [hunk 2]
```diff
@@ -382,8 +390,8 @@ div#grid {
   text-align: center;
   width: 100vw;
   height: 100vh;
-  -webkit-mask-image: radial-gradient(rgba(0, 0, 0, 0.5) 0%, rgba(0, 0, 0, 0.25) 50%, rgba(0, 0, 0, 1) 100%);
-  background-image: url(//ubq.fi/image/grid-1.png);
+  /* -webkit-mask-image: radial-gradient(rgba(0, 0, 0, 0.5) 0%, rgba(0, 0, 0, 0.25) 50%, rgba(0, 0, 0, 1) 100%); */
+  /* background-image: url(//ubq.fi/image/grid-1.png); */
```
Comment (0x4007) [inline pages/styles/index.css]: This shouldn't affect your performance its just a gradient kek

Comment (0x4007) [reply pages/styles/index.css]: Aka you should re-enable

File: pages/styles/index.css [hunk 3]
```diff
@@ -351,7 +359,7 @@ div[id^="inventory-"] figure {
   animation-direction: normal;
   animation-fill-mode: none;
   animation-play-state: running;
-  animation-name: inventory-chevron;
+  /* animation-name: inventory-chevron; */
```
Comment (0x4007) [inline pages/styles/index.css]: Same as my later comments because I did my review from bottom to top

File: pages/styles/index.css [hunk 4]
```diff
@@ -319,17 +323,21 @@ div[id^="inventory-"] svg {
...
-  text-align: center;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  /* width: 1440px; */
+  /* display: inline-block; */
+  /* height: 48px; */
+  /* vertical-align: middle; */
+  /* text-align: center; */
```
Comment (0x4007) [inline pages/styles/index.css]: These need to be on or else this isn't going to render correctly on mobile.

File: pages/styles/index.css [hunk 5]
```diff
@@ -319,17 +323,21 @@ div[id^="inventory-"] svg {
 }
 
 div[id^="inventory-"] {
-  overflow: scroll;
+  overflow: auto;
   -webkit-overflow-scrolling: touch;
-  -webkit-mask-image: linear-gradient(to right, #00000000 0%, #000000ff 12.5%, #000000ff 87.5%, #00000000 100%);
+  /* -webkit-mask-image: linear-gradient(to right, #00000000 0%, #000000ff 12.5%, #000000ff 87.5%, #00000000 100%); */
```
Comment (0x4007) [inline pages/styles/index.css]: Scroll and gradient, yea, for mobile.

File: pages/styles/index.css [hunk 6]
```diff
@@ -296,9 +298,11 @@ input[value*="Connect"]:hover {
 
 #common > div[id^="inventory-"] {
   margin: 0;
+  box-sizing: border-box;
   width: 100%;
   display: block;
-  backdrop-filter: blur(8px);
+  /* backdrop-filter: blur(8px); */
```
Comment (0x4007) [inline pages/styles/index.css]: blur

File: pages/styles/index.css [hunk 7]
```diff
@@ -83,10 +83,12 @@ div[id^="inventory-"] aside,
...
-  display: inline-block;
-  margin: 8px 16px;
-  height: 24px;
-  vertical-align: middle;
+  /* display: inline-block; */
+  /* margin: 8px 16px; */
+  padding: 3px 8px 0px 8px;
+  /* margin-top: 3px; */
+  /* height: 24px; */
+  /* vertical-align: middle; */
```
Comment (0x4007) [inline pages/styles/index.css]: I'm not sure whats going on here looks like some spacing stuff. I suppose you could delete if it looks the same? I see the inventory is shorter compared to production.

File: pages/styles/index.css [hunk 8]
```diff
@@ -33,7 +33,7 @@ header > div {
   position: relative;
   /* box-shadow: 0 0 96px #00ffff20; */
   border: 1px solid #ffffff10;
-  backdrop-filter: blur(24px);
+  /* backdrop-filter: blur(24px); */
```
Comment (0x4007) [inline pages/styles/index.css]: blur

File: pages/styles/index.css [hunk 9]
```diff
@@ -8,7 +8,7 @@ header {
   padding: 8px;
   /* background-color: #000000; */
   border: 1px solid #ffffff10;
-  backdrop-filter: blur(24px);
+  /* backdrop-filter: blur(24px); */
```
Comment (0x4007) [inline pages/styles/index.css]: blur

File: components/ui/widget.tsx [hunk 10]
```diff
@@ -46,3 +47,25 @@ export const PriceExchange = (props: { from: string; to: string; value: number }
...
+export const Loading = (props: { text: string }): JSX.Element => (
+  <div className="h-20 flex items-center justify-center text-lg text-white text-opacity-25">
+    <span className="mr-4">{props.text}</span>
+    <span className="scale-150">{Spinner}</span>
+  </div>
+);
+
+export const Transacting = (): JSX.Element => (
+  <div className="border-accent border bg-accent bg-opacity-10 border-solid absolute top-0 right-0 mr-4 mt-4 rounded-full py-1 px-2 text-accent">
+    Transacting {Spinner}
```
Comment (0x4007) [inline components/ui/widget.tsx]: Common lingo is to say "transaction pending" or "waiting for transaction..." etc.

File: components/ui/icons.tsx [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+);
+
+export const LiquidIcon = (
+  <span className="align-middle">
+    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" className="h-6 mr-2">
+      <path fill="none" d="M0 0h20v20H0z" />
+      <path d="M10 2s-6.5 5.16-6.5 9.5a6.5 6.5 0 1 0 13 0C16.5 7.16 10 2 10 2zm0 14.5c-2.76 0-5-2.24-5-5 0-2.47 3.1-5.8 5-7.53 1.9 1.73 5 5.05 5 7.53 0 2.76-2.24 5-5 5zm-2.97-4.57c.24 1.66 1.79 2.77 3.4 2.54a.5.5 0 0 1 .57.49c0 .28-.2.47-.42.5a4.013 4.013 0 0 1-4.54-3.39c-.04-.3.19-.57.5-.57.25 0 .46.18.49.43z" />
+    </svg>
+  </span>
+);
```
Comment (0x4007) [inline components/ui/icons.tsx:16]: This is pretty nifty

File: components/common.tsx [hunk 12]
```diff
@@ -104,6 +101,12 @@ export function _renderControls() {
     setAccount(await fetchAccount());
   };
 
+  if (!PROD) {
```
Comment (0x4007) [inline components/common.tsx:47]: Here you can load the prod stylesheets etc with blur and video.

**Review by 0x4007 — CHANGES_REQUESTED** [review]
@Zequez awesome work! Last high level comment, but, regarding the table you created I have some adjustments I would like for you to eventually get around to!

image

### Regarding the columns

- `ID` we can hide
- `Pending Reward` is perfect!
- `LP` -> `Deposited` nobody really thinks about their money in LP token terms - you should be able to convert the LP token into its $$$ amount (thats the only thing people care about) using the Curve contract I believe? Ask @zgorizzo69 lol
- `Shares` I think is purely an internal thing that nobody will pay attention to either unless they are debugging something, so you can hide it.
- `Time left` -> `Unlock Time` ideally if you could calculate when the lockup will be available and write the date like 10 Jan 2022 would be awesome!

File: pages/styles/index.css (hunk 1 above)
Comment (Zequez) [reply pages/styles/index.css]: Oh I know, I must have committed it accidentally

**Review by 0x4007 — APPROVED** [review]

**Comment by Zequez** [comment]
Aaaand I think I'm done, it's ready!

**Review by 0x4007 — APPROVED** [review]
Skimmed briefly, lgtm

**Pull request merged**